test/unit/test_jobs.py

```diff
@@ -259,8 +289,13 @@ def test_get_result_twice(self, service):
         job_cls.custom_result = custom_result
 
         job = run_program(service=service, job_classes=job_cls)
-        _ = job.result()
-        _ = job.result()
+        with mock.patch.object(
+            RuntimeJob,
+            "wait_for_final_state",
+            side_effect=self._fake_wait_for_final_state(service, job),
+        ):
```

Comment:
This is too verbose to repeat in all tests, could you move to a function in utils and then use it in tests?

Reply:
Since `mock.patch.object` is used as a context manager how can we move it to a separate function?

Reply:
Could you try if something like this works?
```
with mock_wait_for_final_state(service, job):
    job.wait_for_final_state()
```

```
def mock_wait_for_final_state(service, job):
    return mock.patch.object(
            RuntimeJob,
            "wait_for_final_state",
            side_effect=fake_wait_for_final_state(service, job),
        )
```